File: bundles/gitlab-eclipse-plugin/src/com/gitlab/eclipse/lsp/FeatureStateChange.java
package com.gitlab.eclipse.lsp;

import java.util.List;

public class FeatureStateChange {
	private String featureId;

	private List<FeatureStateChangeCheck> engagedChecks;
	
	public FeatureStateChange(String featureId, List<FeatureStateChangeCheck> engagedChecks) {
		this.featureId = featureId;
		this.engagedChecks = engagedChecks;
	}

	public String getFeatureId() {
		return featureId;
	}

	public List<FeatureStateChangeCheck> getEngagedChecks() {
		return engagedChecks;
	}

	public void setFeatureId(String featureId) {
		this.featureId = featureId;
	}

	public void setEngagedChecks(List<FeatureStateChangeCheck> engagedChecks) {
		this.engagedChecks = engagedChecks;
	}
}

File: gitlab-language-server/src/main/kotlin/com/gitlab/eclipse/lsp/FeatureStateChange.kt
package com.gitlab.eclipse.lsp

class FeatureStateChange(var featureId: String, engagedChecks: List<FeatureStateChangeCheck>) {
    private var engagedChecks: List<FeatureStateChangeCheck>

    init {
        this.engagedChecks = engagedChecks
    }

    fun getEngagedChecks(): List<FeatureStateChangeCheck> {
        return engagedChecks
    }

    fun setEngagedChecks(engagedChecks: List<FeatureStateChangeCheck>) {
        this.engagedChecks = engagedChecks
    }
}